docs/table/construct_table.rst

```diff
@@ -471,6 +471,29 @@ To create a copy of an existing table that is empty (has no rows)::
  float64 float64
  ------- -------
 
+Empty array of a known size
+---------------------------
+
+If you do know the size your table will be, but don't know the values in
+advance, you can create a zeroed numpy array and build the astropy table from
+it::
+
+>>> N = 10
+>>> dtype = [('a', 'i4'), ('b', 'f8'), ('c', 'S2')]
+>>> temp = np.zeros(N, dtype=dtype)
+>>> t = Table(data=temp)
+>>> t
+
+For example, you can then fill in this table row-by-row from values extracted
+from an existing table::
+
+>>> random = np.zeros(100, dtype=dtype)
+>>> random[:] = np.random.randn(*random.shape)
+>>> t2 = Table(data=random)
+>>> for i, x in enumerate(random):
+>>>     if i % 10 == 0:
+>>>         t[i/10] = x[i]
```

Comment:
While I know this does not represent your original use case that resulted in this example, I am not convinced that this example is the best one to put forward, mainly because you can avoid the loop by simply doing `t2[::10]` and not even have to use `t`.

Reply:
I agree that it's not the best example, because of exactly that reason. Do you have a suggestion for a better example to demonstrate my use case?

Reply:
I've written a new example for this part: does it help?